zipkin/src/main/java/zipkin2/Span.java

```diff
@@ -641,7 +641,7 @@ public Span build() {
    */
   public static String normalizeTraceId(String traceId) {
     if (traceId == null) throw new NullPointerException("traceId == null");
-    int length = traceId.length();
```

Comment:
trim has performance impact and this is the lowest level library... seems the trim if occurs should happen in the UI or Query controller instead..

Reply:
so I suggest if this is for query api then maybe add the trim there and in one of the api tests add a test case with the space?

Reply:
I think it's okay to add trim when querying, this performance is not affected at all

Reply:
2、in ZipkinQueryApiV2, we add this “ traceId = traceId != null ? traceId.trim() :null; ”

Reply:
yeah if even possible for that to be null in ZipkinQueryApiV2 which shouldn't be possible. trimming is fine there vs here which is core library used by many instrumentation

Reply:
yes,  ill  pull request  again